### PR TITLE
chore(scripts/lint-packages): add HPND as allowed license

### PR DESCRIPTION
**Pull request by TomJo2000**
We have a lot of packages using the HPND license.
Currently they are all using `custom` for it,
this PR adds `HPND` as an accepted license in the package build linter,
and updates all applicable packages `TERMUX_PKG_LICENSE` field.

This shouldn't necessitate package rebuilds.
If I'm wrong about that, please tell me and I'll split the PR into multiple commits for batching.

<h1><em></em></h1> 

<details><summary>Linter checks were run locally, and all pass.</summary>
<p>

image

</p>
</details>

### CODE REVIEW

**Comment by Grimler91** [comment]
Thanks! Please add HPND to list of licenses in [termux_step_install_license.sh](https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_install_license.sh#L30-L41) as well. This should remove the need to specify TERMUX_PKG_LICENSE_FILE for some of the packages.

**Comment by TomJo2000** [comment]
Derp, missed that one.

**Comment by TomJo2000** [comment]
Just had a read, I think some more extensive refactoring might be in order for that one.
It's currently a very awkward mix of POSIX-isms and Bash specific features as core parts of the script.
I'll see if it fits within this PR or warrants its own.

**Comment by Grimler91** [comment]
Yeah, indeed, large parts of our scripts could benefit from some improvements :) IMO it could be a separate PR.

**Comment by TomJo2000** [comment]
I am gonna split out the `termux_step_install_license` changes into a separate PR since I more or less rewrote the whole script.

Currently doing regression testing locally.

File: packages/poppler-data/build.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 TERMUX_PKG_HOMEPAGE=https://poppler.freedesktop.org/
 TERMUX_PKG_DESCRIPTION="Consists of encoding files for use with poppler"
-# Makefile is licensed under HPND.
-TERMUX_PKG_LICENSE="BSD 3-Clause, GPL-2.0"
+TERMUX_PKG_LICENSE="HPND, BSD 3-Clause, GPL-2.0"
 TERMUX_PKG_LICENSE_FILE="COPYING, COPYING.adobe, COPYING.gpl2"
```
Comment (TomJo2000) [inline packages/poppler-data/build.sh:4]: I don't think we can get rid of `TERMUX_PKG_LICENSE_FILE` here.

File: root-packages/libx86emu/build.sh [hunk 2]
```diff
@@ -1,7 +1,6 @@
 TERMUX_PKG_HOMEPAGE=https://github.com/wfeldt/libx86emu
 TERMUX_PKG_DESCRIPTION="x86 emulation library"
-# License: HPND
-TERMUX_PKG_LICENSE="custom"
+TERMUX_PKG_LICENSE="HPND"
 TERMUX_PKG_LICENSE_FILE="LICENSE, LICENSE_INFO"
```
Comment (TomJo2000) [inline root-packages/libx86emu/build.sh:4]: I don't think we can get rid this instance either.

File: x11-packages/scite/build.sh [hunk 3]
```diff
@@ -1,7 +1,6 @@
 TERMUX_PKG_HOMEPAGE=https://www.scintilla.org/SciTE.html
 TERMUX_PKG_DESCRIPTION="A free source code editor"
-# License: HPND
-TERMUX_PKG_LICENSE="custom"
+TERMUX_PKG_LICENSE="HPND"
 TERMUX_PKG_LICENSE_FILE="scite/License.txt"
```
Comment (TomJo2000) [inline x11-packages/scite/build.sh:4]: This one obviously needs to stay.

**Comment by TomJo2000** [comment]
> Yeah, indeed, large parts of our scripts could benefit from some improvements :) IMO it could be a separate PR.

- #20867

Is now merged, so this PR can move ahead.
If there's no other comments I'd like to merge this in ~4 hours.

**Comment by TomJo2000** [comment]
I decided this should probably be split into 2 commits.

**Pull request merged**